### ISSUE TITLE
Start using Cake.Recipe

### ISSUE THREAD

**Issue by gep13** [cake-contrib/Cake.ResxConverter#3]
@jzeferino @louisfischer are you guys familiar with the package of scripts that I have created called Cake.Recipe?

This is used on almost all the addin's within the Cake-Contrib Organisation, which means there are consistent builds across all the repositories, and when a change is made to Cake.Recipe, it can be applied across all the repositories.

**Comment by luisgoncalves** [comment]
I don't think we are. We'll take a look when we have time.

Does it have any minimum requirements regarding the Git branching model?

**Comment by gep13** [comment]
@luisgoncalves Cake.Recipe works best when you are using GitFlow, which you seem to be set up to use, but it is not a requirement.

Would you be interested in me putting a PR together to add Cake.Recipe to this repo?

**Comment by jzeferino** [comment]
@gep13  can you elaborate a bit more whats the benefits of Cake.Recipe? What are the implications of using it?
You can PR it.

**Comment by gep13** [comment]
@jzeferino sure thing....

Over the course of creating around 10 addins, and then helping to support another 40 within the Cake Contrib Organisation, it became clear that the builds that are being done are almost exactly the same (only a few minor differences between each).  When adding a new step into the build process (for example sending a notification to Gitter or Twitter when a release succeeds), that functionality had to be added to all repositories.  Cake.Recipe is a collection of Cake Build Tasks which are deployed as a NuGet Package, and which is restored when the build starts.  As a result, a new build step (for example, the addition of GitLink to allow source code debugging of addins) can be added in one place, and automatically pushed out to all repositories without needing to change things in lots of places.

Cake.Recipe including all of the major builds steps that are typically required, i.e:

* Build (support for both MSBuild and XBuild)
* Testing (support for NUnit, xUnit, VSTest, Fixie)
* Test Report Generation using ReportUnit
* Code Coverage Generation using OpenCover
* Uploading of Code Coverage results to Coveralls.io
* Analysis (using DupFinder, InspectCode)
* Notifications (Twitter and Gitter)
* Semantic Versioning
* Release Notes generation
* Creation of NuGet and Chocolatey Packages
* Publishing to MyGet, NuGet, Chococolatey
* AppVeyor integrations for pushing artifacts, etc
* Wyam Documentation generation and publishing of site
* Probably a few more things that I can't think of just now

All of the above are configurable in what I refer to as a setup.cake file, so you can choose which ones to run and which ones to not.  The tasks are also based on some conventions, i.e. where you place certain files on the file system within the project, which can also control which tasks are executed.

Please let me know if you have any questions.

**Comment by luisgoncalves** [comment]
So, if I understood correctly, it's just a matter of installing the NuGet package and adding the setup.cake file with the appropriate flags and parameters. Sounds good!

**Comment by gep13** [comment]
@luisgoncalves yes, that pretty much sums it up.  I might have to move some files around on the file system to put them into the correct convention driven folder as well.

**Comment by jzeferino** [comment]
@gep13 feel free to PR with the first implementation. Thanks for the contribution.

**Comment by gep13** [comment]
@jzeferino yes, I haven't forgotten about this, it is on my list of things to do.
